editor/filesystem_dock.cpp

```diff
@@ -1293,33 +1293,66 @@ void FileSystemDock::_try_duplicate_item(const FileOrFolder &p_item, const Strin
 			if (err != OK) {
 				EditorNode::get_singleton()->add_io_error(TTR("Error duplicating:") + "\n" + old_path + ".import\n");
 			}
+		}
+
+		_fix_uid_of_duplicated_item(p_item.is_file, old_path, new_path);
+	} else {
+		EditorNode::get_singleton()->add_io_error(TTR("Error duplicating:") + "\n" + old_path + "\n");
+	}
+}
+
+void FileSystemDock::_fix_uid_of_duplicated_item(bool p_old_is_file, const String &p_old_path, const String &p_new_path) const {
+	if (p_old_is_file) {
+		// Ignore some files, since they do not use a uid
+		String ext = p_new_path.get_extension();
+		// FIXME: Add more file extensions to this list
+		if (ext == "gd" ||
+				ext == "gdshader" ||
+				ext == "txt" ||
+				ext == "md") {
+			return;
+		}
```

Comment:
You could instead check if the resource has UID available:
```C++
ResourceUID::ID uid = ResourceLoader::get_resource_uid(p_old_path);
if (uid == ResourceUID::INVALID_ID) {
    return;
}
```

Reply:
Thank You! This is exactly something I was looking for.